diff -Nur libgeotiff.orig/makefile.vc libgeotiff/makefile.vc
--- libgeotiff.orig/makefile.vc	Thu Mar 29 18:05:21 2012
+++ libgeotiff/makefile.vc	Fri Nov 16 16:21:34 2012
@@ -7,10 +7,14 @@
 # You may want to add optimization options to the CFLAGS macro as well. 
 #
 
-OSGEO4W = c:\OSGeo4W
+LIBTIFF_DIR = ..\libtiff
 
-TIFF_INC = -I$(OSGEO4W)\include
-TIFF_LIB_DLL = $(OSGEO4W)\lib\libtiff_i.lib
+TIFF_INC = -I$(LIBTIFF_DIR)\libtiff
+TIFF_LIB_DLL = $(LIBTIFF_DIR)\libtiff\libtiff_i.lib
+
+PROJ_DIR = ..\proj4
+PROJ_INC = $(PROJ_DIR)\instdir\include
+PROJ_LIB_DLL = $(PROJ_DIR)\instdir\lib\proj.lib
 
 # Installation locations (with install, or devinstall targets)
 PREFIX =	release	
@@ -33,8 +37,14 @@
 DLLNAME = geotiff.dll
 
 # Set optimization or debug flags here.
-CFLAGS  = $(INCL) /MD /Ox /nologo
-#CFLAGS  = $(INCL) /MD /Zi /nologo
+OPTFLAGS = /MD /Ox /nologo
+CFLAGS  = $(INCL) $(OPTFLAGS)
+
+!IFDEF WANT_PROJ4
+CFLAGS = $(CFLAGS) /DHAVE_LIBPROJ=1 /DHAVE_PROJECTS_H=1
+INCL = $(INCL) -I$(PROJ_INC)
+TIFF_LIB_DLL = $(TIFF_LIB_DLL) $(PROJ_LIB_DLL)
+!ENDIF
 
 #
 OBJS    = \
@@ -71,7 +81,7 @@
 
 #
 $(DLLNAME):    $(OBJS)
-	link /dll /out:$(DLLNAME) /implib:geotiff_i.lib $(OBJS) $(TIFF_LIB_DLL)
+	link /dll /out:$(DLLNAME) /implib:geotiff_i.lib $(OBJS) $(TIFF_LIB_DLL) /debug /manifest
 	if exist $(DLLNAME).manifest mt -manifest $(DLLNAME).manifest -outputresource:$(DLLNAME);2
 
 geo_config.h:	geo_config.h.vc
diff -Nur libgeotiff-1.4.0/libxtiff/xtiff.c libgeotiff/libxtiff/xtiff.c
--- libgeotiff-1.4.0/libxtiff/xtiff.c	Wed Oct  5 23:34:23 2011
+++ libgeotiff/libxtiff/xtiff.c	Tue Jan  8 11:46:56 2013
@@ -147,6 +147,17 @@
     return tif;
 }
 
+TIFF*
+XTIFFOpenW(const wchar_t* name, const char* mode)
+{
+    /* Set up the callback */
+    XTIFFInitialize();
+
+    /* Open the file; the callback will set everything up
+     */
+    return TIFFOpenW(name, mode);
+}
+
 TIFF*
 XTIFFFdOpen(int fd, const char* name, const char* mode)
 {
diff -Nur libgeotiff-1.4.0/libxtiff/xtiffio.h libgeotiff/libxtiff/xtiffio.h
--- libgeotiff-1.4.0/libxtiff/xtiffio.h	Wed Oct  5 21:34:23 2011
+++ libgeotiff/libxtiff/xtiffio.h	Sat Nov 17 10:33:57 2012
@@ -67,6 +67,7 @@
 
 extern void CPL_DLL XTIFFInitialize();
 extern TIFF CPL_DLL * XTIFFOpen(const char* name, const char* mode);
+extern TIFF CPL_DLL * XTIFFOpenW(const wchar_t* name, const char* mode);
 extern TIFF CPL_DLL * XTIFFFdOpen(int fd, const char* name, const char* mode);
 extern void CPL_DLL XTIFFClose(TIFF *tif);
 
diff -Nur libgeotiff-1.4.0/libxtiff/xtiffio.h libgeotiff/libxtiff/xtiffio.h
--- libgeotiff-1.4.0/geotiff_proj4.c
+++ libgeotiff/geotiff_proj4.c
@@ -1270,4 +1270,41 @@
     }
 
+    else if( psDefn->CTProjection == CT_HotineObliqueMercatorAzimuthCenter )
+    {
+        /* special case for swiss oblique mercator : see GDAL bug 423 */
+        if( fabs(psDefn->ProjParm[2] - 90.0) < 0.0001
+            && fabs(psDefn->ProjParm[3]-90.0) < 0.0001 )
+        {
+            sprintf( szProjection+strlen(szProjection),
+                     "+proj=somerc +lat_0=%.16g +lon_0=%.16g"
+                     " +k_0=%.16g +x_0=%.16g +y_0=%.16g ",
+                     psDefn->ProjParm[0],
+                     psDefn->ProjParm[1],
+                     psDefn->ProjParm[4],
+                     psDefn->ProjParm[5],
+                     psDefn->ProjParm[6] );
+        }
+        else
+        {
+            sprintf( szProjection+strlen(szProjection),
+                     "+proj=omerc +lat_0=%.16g +lonc=%.16g +alpha=%.16g"
+                     " +k=%.16g +x_0=%.16g +y_0=%.16g ",
+                     psDefn->ProjParm[0],
+                     psDefn->ProjParm[1],
+                     psDefn->ProjParm[2],
+                     psDefn->ProjParm[4],
+                     psDefn->ProjParm[5],
+                     psDefn->ProjParm[6] );
+
+            /* RSO variant - http://trac.osgeo.org/proj/ticket/62 */
+            /* Note that gamma is only supported by PROJ 4.8.0 and later. */
+            /* FIXME: how to detect that gamma isn't set to default value */
+            /*if( psDefn->ProjParm[3] != 0.0 )
+            {
+                sprintf( szProjection+strlen(szProjection), "+gamma=%.16g ",
+                         psDefn->ProjParm[3] );
+            }*/
+        }
+    }
 /* ==================================================================== */
 /*      Handle ellipsoid information.                                   */
